transit_odp/publish/views/timetable/seasonal_services.py

```diff
@@ -64,55 +103,79 @@ def get_form_kwargs(self, step=None):
         return HttpResponseRedirect(
             reverse(
                 "seasonal-service",
-                kwargs={"pk1": org_id},
+                kwargs={"pk1": self.org_id},
                 host=config.hosts.PUBLISH_HOST,
             )
+            + get_table_page(self.page)
         )
 
 
-class SeasonalServiceView(OrgUserViewMixin, SingleTableView):
-    template_name = "publish/seasonal_services/index.html"
+class SeasonalServiceEditDateView(OrgUserViewMixin, BaseUpdateView):
+    template_name = "publish/seasonal_services/edit_date.html"
+    form_class = SeasonalServiceEditDateForm
     model = SeasonalService
-    table_class = SeasonalServiceTable
-    paginate_by = 10
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        self.page = self.request.GET.get("page", None)
+        kwargs.update(
+            {
+                "org_id": self.org_id,
+                "page": self.page,
+            }
+        )
+        return kwargs
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        org_id = self.kwargs["pk1"]
-        context["pk1"] = org_id
-        context["organisation"] = Organisation.objects.get(id=org_id)
-        context[
-            "seasonal_services_counter"
-        ] = SeasonalService.objects.get_count_in_organisation(org_id)
+        context.update({"pk1": self.org_id})
         return context
 
-    def get_table_data(self, **kwargs):
-        org_id = self.kwargs["pk1"]
-        return SeasonalService.objects.filter(licence__organisation_id=org_id)
+    def get_queryset(self):
+        self.org_id = self.kwargs.get("pk1")
+        seasonal_service_id = self.kwargs.get("pk")
+        return SeasonalService.objects.filter(id=seasonal_service_id)
 
+    @transaction.atomic
+    def form_valid(self, form):
+        self.object.start = form.cleaned_data["start"]
+        self.object.end = form.cleaned_data["end"]
+        self.object.save()
+        return HttpResponseRedirect(
+            reverse(
+                "seasonal-service",
+                kwargs={"pk1": self.org_id},
+                host=config.hosts.PUBLISH_HOST,
+            )
+            + get_table_page(self.page)
+        )
 
-class SeasonalServiceEditDateView(OrgUserViewMixin, SingleTableView):
-    template_name = "publish/seasonal_services/index.html"
+
+class SeasonalServiceDeleteView(OrgUserViewMixin, BaseDeleteView):
+    template_name = None
     model = SeasonalService
-    table_class = SeasonalServiceTable
-    paginate_by = 10
 
-    def get_context_data(self, **kwargs):
-        context = super().get_context_data(**kwargs)
-        org_id = self.kwargs["pk1"]
-        context["pk1"] = org_id
-        # uncomment when BODP-5626 merged
-        context[
-            "seasonal_services_counter"
-        ] = 12  # SeasonalService.objects.get_seasonal_service_counter(org_id)
-        return context
+    def post(self, request, **kwargs):
```

Comment:
Probably make this a formview or something. The csrf stuff isnt done here. Create a simple form but dont render it. and do the delete in `def form_valid`

Reply:
Yeah the csrf is done in the `index.html`
to use formview I have to put a pk in the url I understood (you told me) don't do that:
To not insert a  pk in the url when I delete.